teps/0062-catalog-tags-and-hub-categories-management.md

```diff
@@ -0,0 +1,195 @@
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+The Tekton catalog is a collection of blessed Tekton resources that can be used with any system that supports the Tekton API. Each Tekton resource has tags as a field in annotation. All tags in catalog resources are mapped to categories in Hub. Currently when new tags are found in catalog, these tags are mapped to categories called _others_. Hub needs to check every time and do the maintenance of mapping the tag to categories so that it does not fall under Others.
+
+This is [config.yaml](https://github.com/tektoncd/hub/blob/main/config.yaml) in Hub where we map tags to the categories
+
+```yaml
+---
+categories:
+  - name: Build Tools
+    tags: [build-tool]
+  - name: CLI
+    tags: [cli]
+  - name: Cloud
+    tags: [gcp, aws, azure, cloud]
+  - name: Deploy
+    tags: [deploy]
+  - name: Others        👈 Any new tags are mapped here
+    tags: []
+  - name: Test Framework
+    tags: [test]
+```
+
+## Motivation
+
+### Goals
+
+- Make the manual work automated and also properly do the mapping of the resources.
+
+### Non-Goals
+
+### Use Cases (optional)
+
+- Filter the resources on Hub using categories
+- Search the resources based on tags as keywords
+
+## Requirements
+
+- Add category field as annotation in a task
+
+For e.g.
+
+```yaml
+tekton.dev/category: "foo”     👈 Category of a resource
+```
+
+## Proposal
+
+When new tags are added for the resource in the catalog, how these tags should be mapped to categories in Hub. Right now we have to do this manually before every release. As we are doing the catalog refresh at an interval of 30 mins, if this can be automated, will give a better user experience.
+
+For e.g: Consider the following task. It has new tags which are not part of categories of [config.yaml](https://github.com/tektoncd/hub/blob/main/config.yaml)
+
+```yaml
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: pylint
+  labels:
+    app.kubernetes.io/version: '0.2'
+  annotations:
+    tekton.dev/pipelines.minVersion: '0.12.1'
+    tekton.dev/tags: python, pylint   👈 New tags
+    tekton.dev/displayName: pylint
+```
+
+### Proposed Solution (Based on Operator Hub)
+
+- Add categories in catalog task as annotation. There should be only one category for a task. Multiple categories are not allowed.
+
+```yaml
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: pylint
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/category: "foo”     👈 Category of a resource
+    tekton.dev/tags: python, pylint   👈 New tags
+    tekton.dev/displayName: pylint
+```
+
+- CI/Catlin should make sure that category is added for the task and there is only one category. Maintaining the category is the responsibility of the author.
+- If category is not added,Catlin should throw an warning to add a category
```

Comment:
why just a warning and if just a warning should it be `others` as the default fallback (if it is maybe it should be documented?)

Reply:
Since we are making the annotation for category mandatory, I guess we should throw an error, I'll update the pr on that thanks :hugs: